### DISCUSSION TITLE
RabbitMQ client, queue with no consumer not deleted after a recovery and dispose

### DISCUSSION THREAD

**Discussion by rx630** [rabbitmq/rabbitmq-dotnet-client#1612] [Q&A]
Hello,

I use RabbitMQ client v6.8.1 from a docker on a computer A. I connect to the RabbitMQ server 3.11.0 on a computer B.
I have an issue on the following sequence:
- I restart computer B (causing a broker closing connection, well detected by client on computer A)
- When computer B has restarted, computer A automatically reconnects as I configured AutomaticRecoveryEnabled = true and TopologyRecoveryEnabled = true.
- I dispose my consumers (channels) on computer A and create new ones (this is linked to my application, I do it when receiving a message from computer B for a specific reason)
- If I restart again computer B, computer A automatically reconnects again, dispose and create new consumers but there are queues (with no consumers) not deleted on computer B (meaning memory leakage...). 

I don't understand why there are queues with no consumer on the server side as computer A is disposing and creating new consumers. Also all my queues have the flag "Automatic Delete", I suppose the queues with no consumers on the server side never had consumers because it would have been deleted. But why client created queues with no consumers ? Is it linked to the auto recovering ?

Thank you for your help

**Comment by lukebakken** [comment]
@rx630 - the fastest way for me to investigate this issue is for you to provide a runnable console application that reproduces this issue. I appreciate you taking the time to _describe_ what is going on but the definitive way is to provide a runnable application, which should be relatively easy in this case. Thanks.
